### ISSUE TITLE
Inconsistent API: create an experiment

### ISSUE THREAD

**Issue by several27** [mlflow/mlflow#378]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in MLflow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: Linux Ubuntu 16.04 - AWS
- **MLflow installed from (source or binary)**: binary
- **MLflow version (run ``mlflow --version``)**: 0.4.1
- **Python version**: 3.5.2
- **npm version (if running the dev UI): N/A
- **Exact command to reproduce**: N/A

### Describe the problem
The documentation is inconsistent with how the API actually works. The endpoint to create an experiment results with `{"experimentId": X}` instead of `{"experiment_id": X}` as defined in the documentation here: https://www.mlflow.org/docs/latest/rest-api.html#create-experiment

**Comment by mparkhe** [comment]
hi @several27 : Not sure how you are seeing the request structure having `experimentId` field. This is what I see when I `curl` a create experiment request

```
$ > curl -X POST http://localhost:5000/api/2.0/preview/mlflow/experiments/create -d '{"name":"issue_378"}'
{
  "experiment_id": "3"
}
```

Experiment seems to be created correctly:
![image](https://user-images.githubusercontent.com/6900999/44682719-00ca2a00-a9f9-11e8-84e0-32b173d810a7.png)

Can you upload an example of what you are seeing in the request object?

Thanks,
mani

**Comment by several27** [comment]
Hi @mparkhe, that's very strange, is it possible that I'm using some old version of mlflow then? I'm writing a Scala API for mlflow for an internal project, so my first thought was that it's scala fault, but curl returns exactly same results. Below are exact requests I'm running, and the exact responses I'm getting:

```
➜  ~ curl -X POST http://13.57.143.XXX:5000/api/2.0/preview/mlflow/experiments/create -d '{"name":"test123"}'
{
  "experimentId": "16"
}%
```

Other API points correctly return `experiment_id`:
```
➜  ~ curl http://13.57.143.XXX:5000/api/2.0/preview/mlflow/experiments/list
{
  "experiments": [
    {
      "name": "test123",
      "experiment_id": "16",
      "artifact_location": "s3://XXX/experiments/16"
    },
  ]
}%
```

**Comment by mparkhe** [comment]
@several27 : I see. Just took a look at the [change log](https://github.com/mlflow/mlflow/blob/master/CHANGELOG.rst). This was fixed in `0.4.2`. I would recommend moving to a newer version.

mani
